### PR TITLE
feature(tests): basic tests

### PR DESCRIPTION
**Pull request by nappynapster**
@plentymarkets/team-terra

#### Must be done by Terra

Testing

-   [ ] Person 1 (mandatory)

    Browser-Support (relevant for changes in scss / appearance of components)

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

-   [ ] Person 2 (mandatory)

    Browser-Support

    -   [ ] Chrome
    -   [ ] Firefox
    -   [ ] Safari

    Responsiveness

    -   [ ] Desktop
    -   [ ] Tablet
    -   [ ] Mobile

Information transfer

-   [ ] Inform Ceres about changes in `TerraFormComponent` (optional: relevant for changes in this component)

Terra Basic Plugin

-   [ ] Adapt changes (optional: relevant for version updates, global design changes, l10n)

---

#### Must be done by every developer

Please inform a member of Terra (@plentymarkets/team-terra) to get the upper part of the checklist done (with urgency or deadline).

Documentation

-   [ ] Changelog (optional: relevant for every change which influences the API)
-   [ ] Example (updated or created)
-   [ ] JSDoc (optional: relevant for every change which influences the API)
-   [ ] [Google spreadsheet](https://docs.google.com/spreadsheets/d/1OINnux8TEoitV-qdAxqUQaf9oGI_fqFVwfRWenRFfhI/edit#gid=0) (relevant when deprecating public APIs or features)

Testing

-   [ ] Unit Test (updated or created)

### CODE REVIEW

File: src/lib/components/forms/form/form-components/text-input/text-input.interface.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ export interface TextInputInterface extends TerraFormComponentBaseInterface {
     isReadonly: boolean;
     minLength: number;
     maxLength: number;
+    value: string;
```
Comment (nappynapster) [inline src/lib/components/forms/form/form-components/text-input/text-input.interface.ts:9]: habe das hinzugefügt weil das alte text input die TerraInputComponent implementiert hat

File: src/lib/components/forms/form/form-components/text-input/text-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public name: string;
+
+    @Input()
+    public tooltipPlacement: string;
+
+    @Input()
+    public tooltipText: string;
+
+    @Input()
+    public value: string;
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/text-input/text-input.component.ts:48]: Das sollte kein Input sein.

File: src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        });
+    });
+
+    it('should update the value of the input when writing a new value via `writeValue()`', async () => {
+        const value: string = 'my test string';
+        component.writeValue(value);
+        expect(component.value).toEqual(value);
+
+        fixture.detectChanges();
+        expect(await input.getValue()).toEqual(value.toString());
```
Comment (pweyrich) [inline src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts:136]:
```suggestion
        expect(await input.getValue()).toEqual(value);
```
value ist doch schon ein string^^

File: src/lib/components/forms/form/form-components/text-input/text-input.interface.ts (hunk 1 above)
Comment (pweyrich) [reply src/lib/components/forms/form/form-components/text-input/text-input.interface.ts:9]: ich denke das ist nicht notwendig. Haben wir bei den anderen auch nicht gemacht. und es sollte auch kein `@Input` sein.

File: src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts (hunk 3 above)
Comment (nappynapster) [reply src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts]: jap 🙉

File: src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        fixture.detectChanges();
+        expect(await input.getValue()).toEqual(value);
+    });
+
+    it('should call registered change callback whenever the value of the input is changed by the user', async () => {
+        const onChangeCallback: jasmine.Spy = jasmine.createSpy('onChange');
+        component.registerOnChange(onChangeCallback);
+
+        const value: string = 'my test string';
+        await input.setValue(value.toString());
```
Comment (niwolf) [inline src/lib/components/forms/form/form-components/text-input/text-input.component.spec.ts:143]: Ist ja bereits nen string, oder?
```suggestion
        await input.setValue(value);
```

**Review by niwolf — COMMENTED** [review]
Sonst alles top 😄

**Review by niwolf — APPROVED** [review]

**Review by zhajjhassan — APPROVED** [review]

**Review by pweyrich — APPROVED** [review]
Sieht soweit gut aus 👍

**Pull request merged**